src/main/java/com/exactpro/th2/common/event/Event.java

```diff
@@ -302,16 +343,93 @@ protected String formatStateException(String fieldName, Object value) {
         return fieldName + " in event '" + id + "' already sed with value '" + value + '\'';
     }
 
-    @NotNull
+    /**
+     * @deprecated use {@link #getAggregatedStatus} instead
+     */
+    @Deprecated(forRemoval = true)
     protected Status getAggrigatedStatus() {
+        return getAggregatedStatus();
+    }
+
+    @NotNull
+    protected Status getAggregatedStatus() {
         if (status == Status.PASSED) {
-            return subEvents.stream().anyMatch(subEvent -> subEvent.getAggrigatedStatus() == Status.FAILED)
+            return subEvents.stream().anyMatch(subEvent -> subEvent.getAggregatedStatus() == Status.FAILED)
                     ? Status.FAILED
                     : Status.PASSED;
         }
         return Status.FAILED;
     }
 
+    private void batch(int maxEventBatchContentSize, List<EventBatch> result, Map<EventID, List<com.exactpro.th2.common.grpc.Event>> eventGroups, @Nullable EventID eventID) throws JsonProcessingException {
+        eventID = ObjectUtils.defaultIfNull(eventID, DEFAULT_EVENT_ID);
```

Comment:
Can be replaces with `Objects.requireNonNullElse`.
Let's start using Java standard library methods)